### PR TITLE
Added tests for common utilities CollectionUtil & HtmlUtils

### PR DESCRIPTION
**Pull request by freddytuxworth**
Hi,
I've analysed this codebase and written a few tests using Diffblue Cover to fill some gaps in coverage. Hopefully these tests will allow you to spot any regressions caused by future code changes.
Please let me know if I can help with any more tests.

### CODE REVIEW

File: common/src/test/java/org/keycloak/common/util/CollectionUtilTest.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+  @Test
+  public void joinInputNoneSeparatorNullOutputEmpty() {
+    final ArrayList<String> strings = new ArrayList<String>();
+    final String retval = CollectionUtil.join(strings, null);
+    Assert.assertEquals("", retval);
+  }
+
+  @Test
+  public void joinInputNoneSeparatorNullOutputEmpty() {
```
Comment (JohnLBergqvist) [inline common/src/test/java/org/keycloak/common/util/CollectionUtilTest.java]: Duplicate test, remove.

**Review by JohnLBergqvist — CHANGES_REQUESTED** [review]

File: common/src/test/java/org/keycloak/common/util/CollectionUtilTest.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+import java.lang.reflect.Array;
+import java.util.ArrayList;
+
+public class CollectionUtilTest {
+
+  @Rule
+  public ExpectedException thrown = ExpectedException.none();
+  @Rule
+  public Timeout globalTimeout = new Timeout(10000);
```
Comment (sudo-jaa) [inline common/src/test/java/org/keycloak/common/util/CollectionUtilTest.java]: I would remove this timeout rule and the org.junit.rules.Timeout import at line 7. 

Timeout rules are something we added in at the request of a customer to prevent any hanging tests during execution. With a small amount of tests like this, it's better for the process to cut out as much of the unnecessary stuff as possible.

File: common/src/test/java/org/keycloak/common/util/CollectionUtilTest.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+  @Test
+  public void joinInputNoneOutputEmpty() {
+    final ArrayList<String> strings = new ArrayList<String>();
+    final String retval = CollectionUtil.join(strings, ",");
+    Assert.assertEquals("", retval);
+  }
+
+  @Test
+  public void joinInput2SeparatorNull() {
```
Comment (sudo-jaa) [inline common/src/test/java/org/keycloak/common/util/CollectionUtilTest.java:23]: joinInput2SeparatorNull() -> joinInputSeparatorNull()

File: common/src/test/java/org/keycloak/common/util/HtmlUtilsTest.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+import org.junit.rules.ExpectedException;
+import org.junit.rules.Timeout;
+import org.keycloak.common.util.HtmlUtils;
+
+public class HtmlUtilsTest {
+
+  @Rule
+  public ExpectedException thrown = ExpectedException.none();
+  @Rule
+  public Timeout globalTimeout = new Timeout(10000);
```
Comment (sudo-jaa) [inline common/src/test/java/org/keycloak/common/util/HtmlUtilsTest.java]: Both these rules and their associated imports can be removed from this test class

File: common/src/test/java/org/keycloak/common/util/NetworkUtilsTest.java [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+import java.lang.reflect.Method;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+
+public class NetworkUtilsTest {
+
+  @Rule
+  public ExpectedException thrown = ExpectedException.none();
+  @Rule
+  public Timeout globalTimeout = new Timeout(10000);
```
Comment (sudo-jaa) [inline common/src/test/java/org/keycloak/common/util/NetworkUtilsTest.java]: Both these rules and their associated imports can be removed from this test class

**Review by sudo-jaa — CHANGES_REQUESTED** [review]
A couple of changes requests: namely removal of unused imports and rules

File: common/src/test/java/org/keycloak/common/util/CollectionUtilTest.java (hunk 3 above)
Comment (petersear) [reply common/src/test/java/org/keycloak/common/util/CollectionUtilTest.java:23]: ?? doesn't the "2" refer to 2 input strings?

File: common/src/test/java/org/keycloak/common/util/CollectionUtilTest.java [hunk 6]
```diff
@@ -0,0 +1,47 @@
+package org.keycloak.common.util;
+
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
```
Comment (petersear) [inline common/src/test/java/org/keycloak/common/util/CollectionUtilTest.java:6]: ExpectedException input can be removed as there are no exception tests (remaining) in this class.

File: common/src/test/java/org/keycloak/common/util/CollectionUtilTest.java [hunk 7]
```diff
@@ -0,0 +1,47 @@
+package org.keycloak.common.util;
+
+import org.junit.Assert;
+import org.junit.Rule;
```
Comment (petersear) [inline common/src/test/java/org/keycloak/common/util/CollectionUtilTest.java:4]: Rule import not needed.

File: testsuite/integration-arquillian/tests/base/src/main/java/org/keycloak/testsuite/adapter/page/PhotozClientAuthzTestApp.java [hunk 8]
```diff
@@ -159,7 +159,7 @@ public void requestEntitlement(JavascriptStateValidator validator) {
     
     public void requestEntitlements(JavascriptStateValidator validator) {
         testExecutor.executeAsyncScript("var callback = arguments[arguments.length - 1];" +
-                "window.authorization.entitlement('photoz-restful-api').then(function (rpt) {" +
+                "window.authorization.entitlement('photoz-restful-api', {}).then(function (rpt) {" +
```
Comment (petersear) [inline testsuite/integration-arquillian/tests/base/src/main/java/org/keycloak/testsuite/adapter/page/PhotozClientAuthzTestApp.java:162]: What is this change related to?

File: common/src/test/java/org/keycloak/common/util/HtmlUtilsTest.java [hunk 9]
```diff
@@ -0,0 +1,20 @@
+package org.keycloak.common.util;
+
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
```
Comment (petersear) [inline common/src/test/java/org/keycloak/common/util/HtmlUtilsTest.java:6]: Remove. No exception tests in this class.

**Review by petersear — CHANGES_REQUESTED** [review]
